plugin/utils/shell.py

```diff
@@ -86,6 +87,183 @@ def create_lods():
 	return msgs
 
 
+def generate_rig_edit():
+    """Automatic rig edit generator by NDP. Detects posed bones and automatically generates nodes and offsets them."""
+    msgs = []
+    #logging.info(f"Generating rig edit")
+    print("-------------------------------------------------------------")
+    print("Initiating rig edit")
+    
+    #Check if selected object is a valid armature
+    if bpy.context.active_object.type != 'ARMATURE':
+        #Object is not an armature. Cancelling.
+        print("No armature selected.")
+        msgs.append(f"No armature selected.")
+        return msgs
+    
+    #Store current mode
+    Mode = bpy.context.mode
```

Comment:
always make normal variables lowercase only in python

Reply:
changed Mode to original_mode and Edit to editnumber